### ISSUE TITLE
Some songs won't download

### ISSUE THREAD

**Issue by Deemitry88** [Malopieds/InnerTune#170]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/z-huang/InnerTune/actions).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce the bug

1. Go to a playlist
2. Sometimes some songs wont download 

### Expected behavior

The songs should download...

### Actual behavior

...but they don't

### Screenshots/Screen recordings

![Screenshot_20240706_011654_InnerTune](https://github.com/Malopieds/InnerTune/assets/91915462/504f90d3-4b19-4f2d-af10-108ec0afe917)

I just get this error that says "The download failed"

### Logs

I don't have access to a pc right now

### InnerTune version

0.5.7

### Android version

Android 12

**Comment by Malopieds** [comment]
> Sometimes some songs wont download

Sometimes as in some will **never** download or some will have to be downloaded again?
Either way I can download both of them

**Comment by Deemitry88** [comment]
They will **never** download, if I try to download them again they will just try to download indefinitely until I stop the download

**Comment by Malopieds** [comment]
> They will never download, if I try to download them again they will just try to download indefinitely until I stop the download

Weird, even when only downloading one? I will need some log for this

**Comment by Deemitry88** [comment]
How do I get thoses logs ? Even If I do `logcat | grep com.malopieds.innertune` I won't get any of innertune's logs but only android's one referencing innertune

**Comment by Malopieds** [comment]
Try the latest debug version, there should be the error in the notification (but the text might be too long) and the error on the logs under the tag `Download failed`, send them here !

**Comment by Malopieds** [comment]
Well I don't know why it doesn't want to build, it works well for me with the same command... You can clone the repo and try it yourself if you know how to do it. Ask me if you have any question

**Comment by Deemitry88** [comment]
I got this error while trying to build
[logErrors.txt](https://github.com/user-attachments/files/16117236/logErrors.txt)

**Comment by Malopieds** [comment]
To build the debug version use this command:
```./gradlew assembleDebug lintFossDebug testFossDebugUnitTest --stacktrace -DskipFormatKtlint```
the output file will be in `app/build/outputs/apk/foss/debug/app-foss-debug.apk`

**Comment by Deemitry88** [comment]
I can figure out how to make it work

**Comment by Malopieds** [comment]
> I can figure out how to make it work

To make it build or to be able to download it?

**Comment by malisipi** [comment]
I think the reason is the rate limit, caching and the following issue. https://github.com/Malopieds/InnerTune/issues/178#issuecomment-2221804917

**Comment by ElhamAryanpur** [comment]
Yeah same issue happened to me, it straight up doesn't even play some songs in the playlist. I agree it might be ratelimit

**Comment by Deemitry88** [comment]
Yeah, it is definitely rate limit, is there any way to wipe the cache so we can download those song again ?

**Comment by Irobrine** [comment]
Seems to be happening to me as well. The longer the song, the slower the download becomes until it basically doesn't download at all. 
Songs below 8 minutes: Max speed 
Songs between 8 to 15 minutes: ~2MB/s
Songs above 15: ~20KB/s
(These are estimates)

**Comment by Deemitry88** [comment]
@Malopieds ?

**Comment by Malopieds** [comment]
I don't know what I can possibly do for a rate limit as of now.

**Comment by ElhamAryanpur** [comment]
I have been trying to play the same songs after weeks, still same issue. possibly not a rate limit in this case? Because even with VPN, without VPN, with or without large time past since last download, the result is the same, unplayable for some songs.

**Comment by Deemitry88** [comment]
> I don't know what I can possibly do for a rate limit as of now.

As I said you could just add the ability to wipe the cache so we can download the songs again

**Comment by Malopieds** [comment]
> As I said you could just add the ability to wipe the cache so we can download the songs again

A rate limit and the cache have nothing to do in common. It may be related to the cache but even re-syncing the song (there is a button for that now) doesn't seems to do anything. It looks like a null pointer but I don't know from where.
